src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -149,6 +155,106 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                     }
                 }
 
+                var nuGetlockFilePath = NuGetLockFileUtilities.GetNuGetLockFilePath(_request.Project);
+                NuGetLockFile nuGetLockFile = null;
+
+                var restorePackagesWithLockFile = _request.Project.RestoreMetadata?.RestorePackagesWithLockFile;
+
+                if (!MSBuildStringUtility.IsTrueOrEmpty(restorePackagesWithLockFile) && File.Exists(nuGetlockFilePath))
+                {
+                    // invalid input since packages.lock.json file exists along with RestorePackagesWithLockFile is set to false.
+                    await _logger.LogAsync(RestoreLogMessage.CreateError(NuGetLogCode.NU1005, Strings.Error_InvalidLockFileInput));
+
+                    // Write the logs into the assets file
+                    var logs = _logger.Errors
+                        .Select(l => AssetsLogMessage.Create(l))
+                        .ToList();
+
+                    LockFile newAssetsFile = null;
+                    if (_request.ExistingLockFile != null)
+                    {
+                        newAssetsFile = _request.ExistingLockFile.Clone();
+                        newAssetsFile.LogMessages = logs;
+                    }
+
+                    return new RestoreResult(
+                                success: false,
+                                restoreGraphs: new List<RestoreTargetGraph>(),
+                                compatibilityCheckResults: new List<CompatibilityCheckResult>(),
+                                msbuildFiles: new List<MSBuildOutputFile>(),
+                                lockFile: newAssetsFile,
+                                previousLockFile: _request.ExistingLockFile,
+                                lockFilePath: _request.ExistingLockFile?.Path,
+                                cacheFile: null,
```

Comment:
I dont follow what consistency? they are not applicable the moment we're out of NoOp check and I'd like to actually pass as `null` then just to avoid any implication what so ever...

Reply:
I don't like the practice where we have 5 different exit points in a 100 line method. 

It's confusing and difficult to follow. Selectively omitting certain values in the result also makes it hard to read. 

The implication is that the file will get written out and you get consistency among restores.